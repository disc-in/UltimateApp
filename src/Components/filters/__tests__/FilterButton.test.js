import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import FilterButton from '../FilterButton';

describe('<FilterButton />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<FilterButton title="OK" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when active', () => {
    const { toJSON } = render(<FilterButton title="OK" active />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls delegates onPress prop', async () => {
    const onPress = jest.fn();

    const { getByText } = render(<FilterButton title="Easy" onPress={onPress} />);

    await fireEvent.press(getByText('Easy'));

    expect(onPress).toHaveBeenCalled();
  });
});
