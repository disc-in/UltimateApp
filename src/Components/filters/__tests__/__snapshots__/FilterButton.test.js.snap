// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#e5e5e5",
      "borderWidth": 1,
      "flexBasis": "30%",
      "flexGrow": 1,
      "marginRight": 5,
      "opacity": 1,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "textAlign": "center",
        "textTransform": "capitalize",
      }
    }
  >
    OK
  </Text>
</View>
`;

exports[`<FilterButton /> renders correctly when active 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#eee",
      "borderColor": "#d5d5d5",
      "borderWidth": 1,
      "flexBasis": "30%",
      "flexGrow": 1,
      "marginRight": 5,
      "opacity": 1,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "textAlign": "center",
        "textTransform": "capitalize",
      }
    }
  >
    OK
  </Text>
</View>
`;
